### DISCUSSION TITLE
How to add a sql connection string to an existing database?

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#2848] [Q&A] [answered]
I'm using preview 4 and trying to add a sql connection to an existing database (happens to be an Azure VM).

I've added 

`var sqlConnection = builder.AddConnectionString("sql", "SQL_CONNECTION");`

And then on my project:

```
var web = builder.AddProject<Projects.AspireTest>("aspire")
	.WithReference(sqlConnection)
```

This is fine running locally, reading the connection string from  the AppHost secrets.

However, if you do an azd deploy, or run azd in your Azure DevOps pipeline, it asks for this connection string as a parameter. e.g. this in the prompt after 'azd deploy':

? Enter a value for the 'sql' infrastructure secured parameter:

I was hoping instead it would use an environment variable (e.g. SQL_CONNECTION), I'm not sure how I supply this parameter at build time in the Azure devops pipeline?

Is there a way to tell it to get this connection string from environment variable instead (was hoping it would use 'SQL_CONNECTION' from above)?

Or am I just going about this the wrong way?

**Comment by davidfowl** [comment]
See https://learn.microsoft.com/en-us/dotnet/aspire/whats-new/preview-4#prompting-for-parameters

**Comment by mip1983** [comment, marked as answer]
Thanks, for that, got this all working. As a small thing, it would be good if there where a prompt some where, perhaps in the `<summary>` comment on 'AddConnectionString', or on azd where it says 'enter a value for sqlConnection' perhaps it could output '(alternatively create an environment variable called AZURE_SQL_CONNECTION)'.

Because the second parameter to AddConnectionString is 'environmentVariableName', I thought that was what it would be looking for as an input, but I guess it's just what the output variable name is on the container app. Maybe that argument should also determine what azd calls it in the main.parameters.json?

(btw: I like the 'enrich' change as a way of augmenting the dbContext, prefer that to how it was)

**Reply by davidfowl** [reply]
> on azd where it says 'enter a value for sqlConnection' perhaps it could output '(alternatively create an environment variable called AZURE_SQL_CONNECTION)'.

Yes this is where it belongs. cc @vhvb1989 

Can you file an issue here for it https://github.com/Azure/azure-dev/issues, this is a small doable change.

> Because the second parameter to AddConnectionString is 'environmentVariableName', I thought that was what it would be looking for as an input, but I guess it's just what the output variable name is on the container app. Maybe that argument should also determine what azd calls it in the main.parameters.json?

That's not a bad idea, we would need to expose this in the manifest. Today there's no coupling between the parameter name and what the tool uses to prompt. 

> (btw: I like the 'enrich' change as a way of augmenting the dbContext, prefer that to how it was)

👍🏾

**Comment by mip1983** [comment]
Issue filed here: https://github.com/Azure/azure-dev/issues/3597
